database/service_db.go

```diff
@@ -69,16 +68,13 @@ func (d *LevelDBServiceDB) All() ([]*service.Service, error) {
 		iter     = d.db.NewIterator(nil, nil)
 	)
 	for iter.Next() {
-		s, err := d.unmarshal(iter.Key(), iter.Value())
+		hash := hash.Hash(iter.Key())
+		s, err := d.unmarshal(hash, iter.Value())
 		if err != nil {
 			// NOTE: Ignore all decode errors (possibly due to a service
 			// structure change or database corruption)
-			if decodeErr, ok := err.(*DecodeError); ok {
-				logrus.WithField("service", decodeErr.hash.String()).Warning(decodeErr.Error())
-				continue
-			}
-			iter.Release()
-			return nil, err
+			logrus.WithField("service", hash.String()).Warning(err.Error())
```

Comment:
This is a change of behavior and now all the unmarshal error are just as a warning and will never be returned.
Is there any reason for this change of behavior?